### PR TITLE
doc(BFormGroup): Update docs and component data

### PR DESCRIPTION
**Pull request by dwgray**
# Describe the PR

Complete the docs and component data for BFormGroup, I've also updated the parity report.

I believe this component may need a rework before we can call it complete -  at minimum, we should consider using script setup and creating a typescript interface. @VividLemon - Do you agree? If so, should I create an issue for this? I do have a note in the parity report, but I think it would be a good idea to track significant changes to the code with issues...

## Small replication

N/A

## PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix :bug: - `fix(...)`
- [ ] Feature - `feat(...)`
- [ ] ARIA accessibility - `fix(...)`
- [x] Documentation update - `docs(...)`
- [ ] Other (please describe)

**The PR fulfills these requirements:**

- [x] Pull request title and all commits follow the [**Conventional Commits**](https://www.conventionalcommits.org/) convention or has an [**override**](https://github.com/googleapis/release-please#how-can-i-fix-release-notes) in this pull request body **This is very important, as the `CHANGELOG` is generated from these messages, and determines the next version type. Pull requests that do not follow conventional commits or do not have an override will be denied**

### CODE REVIEW

File: apps/docs/src/data/components/formGroup.data.ts [hunk 1]
```diff
@@ -7,99 +8,207 @@ export default {
...
         },
       },
       emits: [],
-      slots: [],
+      slots: [
+        {
+          name: 'default',
+          description: 'Content (form checkboxes) to place in the form checkbox group',
+          scope: [
+            {
```
Comment (dwgray) [inline apps/docs/src/data/components/formGroup.data.ts:168]: @VividLemon I'm wondering if it's worth extending the typing work that we (well, mostly you) did for props to cover emits and slots. It feels weird to have the scoped slots in the old format even though they are shaped the same as properties. If so, I'll take that on soon (I want to be judicious about how much time I'm spending on infrastructure vs. getting through the parity assessment/updates - so I'm going to aim for getting a few components covered between diving into infrastructure - we'll see how that goes)

Comment (VividLemon) [reply apps/docs/src/data/components/formGroup.data.ts:168]: Slot scopes and emits aren't really shared, they're more commonalities

**Review by VividLemon — APPROVED** [review]

**Comment by VividLemon** [comment]
> I believe this component may need a rework before we can call it complete - at minimum, we should consider using script setup and creating a typescript interface. @VividLemon - Do you agree? If so, should I create an issue for this? I do have a note in the parity report, but I think it would be a good idea to track significant changes to the code with issues...

I'll do this later. It's not a trivial refactor

**Comment by dwgray** [comment]
> I'll do this later. It's not a trivial refactor. At the time of looking at it I didn't know about `createReusableTemplate`, which simplifies things

Yeah - I'm not planning on doing major reworks of components as part of the docs/parity pass, nor am I expecting you to. 

This is more a question of how you'd like them tracked. Issues or just notes in the parity report?

The secondary question is, where is this component wrt blocking a v1 release? My thought is that for v1, the most important thing is that the API is stable to the best of our ability. e.g. we wouldn't expect sweeping breaking changes until some point (hopefully far in the future) when we would consider a v2. On that level, the only thing that I'd consider changing to get to v1 is the typing of the properties that are currently generically strings but in other places in the code are types eg: `export type Breakpoint = 'sm' | 'md' | 'lg' | 'xl' | 'xxl'` - I think I can fix those without any kind of refactor, but haven't really played with that yet. If you agree, I'll queue that up and handle it once I've got a bit more of the parity report under my belt and have a better idea of the patterns.

**Pull request merged**